Documentation/PropertyWrapper/Parameter.md

```diff
@@ -283,7 +281,7 @@ Exposing the `Component` on a web socket interface requires the client to send t
 ## Path Parameters
 
 Some middleware types and protocols can expose parameters as part of the endpoint defining characteristics. E.g. RESTful and OpenAPI based APIs use the URI path to define endpoints including variables such as `/birds/BIRD_ID` to identify the requested `Bird` by its identifier `BIRD_ID`.
-The `@Parameter` property wrapper exposes the option to explicitly defining a parameter as defining an endpoint using the `.identifier` option.
+We can infer that a specific parameter is part of this path for example: if it contains the word ID in the name and is at the start of the Component. 
```

Comment:
Am I right by the assumption that this would imply the following inference rules (for HTTP):

1. If wrapped property's name contains substring `ID` AND is the first property of the Component => `HTTPParameterMode.path`
2. If not primitive type (or `LosslessStringConvertible`) => `HTTPParameterMode.body`
3. Else => `HTTPParameterMode.query`

Do you think that the potential problems I collected in #18 for this "pattern matching" approach are neglectable (e.g., misspelling, mispositioning)? Could we also consider to use the "conformance to `Identifiable`" (or some other protocol) information as a maybe less error-prone proxy?

Reply:
Thanks that's a good summarization of what I had in mind.
I think the idea of making it depend on identifiable might be a good idea. 

Mispositioning is I think definitely negligible
I think spelling errors should also be uncommon for something like `id`, but perhaps still somewhat error-prone. I think that we are far from done defining these inference rules, so please continue giving us great ideas on how to make it better and more reliable